File: modulo1/segmento3/apunte8EstrDeDatos/ahorcado.py

def generarBaseDatos():
	seguir="si"
	listas=[]

	while seguir=="si":
		lista=input("Ingrese las palabras para jugar: ")
		listas.append(lista)

		seguir=input("Desea agregar otra palabra(si/no)")
	return listas

def sortear(lista):
	from random import randrange
	random = randrange(len(lista))
	return listaJuego[random]

#MAIN
print("*************AHORCADO********")
print("-----------------------------")
print("-----------------------------")
listaJuego=generarBaseDatos()

palabra=sortear(listaJuego)
#CONVIERTO PALABRA EN LISTA
Palabra=list(palabra)
#INICIALIZO VARIABLES 
vidas=6
faltan=len(Palabra)
PalabraMostrar=[]

#CREO LISTA VACIA
for i in range(len(Palabra)):
	PalabraMostrar.append('-')

while vidas>0 and faltan>0:
	print(PalabraMostrar) #muestro lista vacia
	letra=input("Ingrese una letra: ")
	
	if letra in Palabra:
		faltan=faltan - Palabra.count(letra) # resto la cantidad de veces que aparece la letra
		print("Letras Restantes:",faltan)
		
		for i in range(len(Palabra)):
			if Palabra[i] == letra:
				PalabraMostrar[i] = letra
	else:
		vidas= vidas-1
		print("Vidas: " , vidas)
	
if faltan==0:
	print("***********GANASTE!!!!!!")
	print("La palabra es :", palabra)
elif vidas==0:
	print("Perdio!!!")

File: modulo1/segmento3/apunte8EstrDeDatos/Ahorcado.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Random;
import java.util.Collections;


public class Ahorcado{

    public static ArrayList<String> generarBaseDatos(){
        Scanner teclado = new Scanner(System.in);
        String seguir="si";
        String palabras;
        ArrayList<String> listaPalabras  = new ArrayList<String>();

        while(seguir.equals("si")){
            System.out.print("Ingrese las palabras para jugar: ");
            palabras=teclado.nextLine();
            listaPalabras.add(palabras);

            System.out.print("Desea agregar otra palabra(si/no): ");
            seguir= teclado.nextLine();
        }
        return listaPalabras;
    }

    public static String sortear(ArrayList<String>listaJuego){
        Random palabraAleatoria = new Random();
        int n = palabraAleatoria.nextInt(listaJuego.size());
        //System.out.println(listaJuego.get(n));
        return listaJuego.get(n);
    }
    
    public static ArrayList<String> convertStringToArraylist(String palabra) {
        ArrayList<String> stringList = new ArrayList<String>();      
        stringList=Arrays.asList(palabra);
        // for(int i = 0; i<palabra.length();i++){
        //     stringList.add(palabra.(i));
        // }
        return stringList;
    }

    public static void main(String args[]){
        Scanner teclado = new Scanner(System.in);
        //Genero lista de palabras
        ArrayList<String> listaJuego  = new ArrayList<String>();
        listaJuego=generarBaseDatos();

        //Sorteo la palabra de la lista ingresada
        String palabra;
        palabra=sortear(listaJuego);
        System.out.println(palabra);

        //Convierto palabra en lista para ir comparando
        ArrayList<String> palabraLista  = new ArrayList<String>();
        palabraLista=convertStringToArraylist(palabra);

        //Inicializo variables
        Integer vidas = 6;
        Integer faltan= palabraLista.size();
        System.out.println(faltan);
        ArrayList<String> palabraMostrar  = new ArrayList<String>();
        
        //CREO LISTA VACIA
        for(int i=0;i<palabraLista.size();i++){
            palabraMostrar.add("-");
        }
        System.out.println(palabraMostrar);

        while(vidas>0 && faltan>0){
            System.out.println(palabraMostrar);//LISTA VACIA
            System.out.print("Ingrese una letra: ");
            String letra=  teclado.nextLine();
            System.out.println(faltan);
            System.out.println(letra);
            System.out.println(palabraLista);
            if (palabraLista.contains("o")){
                faltan=faltan - Collections.frequency(palabraLista,letra);
                System.out.print("Letras Restantes: "+ faltan);

                // for(i=0;i<palabraLista.size();i++){
                //     if()
                // }
            }

        }

    }

}
